File: extract.js
const fs = require('fs');
const path = require('path');

const tables = [
  'MarketingContact', 'MarketingList', 'MarketingContactList', 'MarketingTag', 
  'MarketingContactTag', 'MarketingSegment', 'EmailTemplate', 'CampaignMetricDaily', 
  'CampaignMetricHourly', 'DomainProviderMetric', 'AudienceMetricDaily', 'LinkMetric', 
  'DeliverabilityAlert', 'MtaNode', 'MtaProviderState'
];

const lines = fs.readFileSync('full_schema.sql', 'utf16le').split('\n');

const out = [];
let in_table = false;
let in_index = false;
let in_fk = false;

for (let i = 0; i < lines.length; i++) {
  const line = lines[i] + '\n';
  
  if (line.startsWith('CREATE TABLE')) {
    const tableName = line.split('"')[1];
    if (tables.includes(tableName)) {
      in_table = true;
      out.push(line);
    } else {
      in_table = false;
    }
  } else if (line.startsWith('CREATE UNIQUE INDEX') || line.startsWith('CREATE INDEX')) {
    const tableName = line.split('"')[3];
    if (tables.includes(tableName)) {
      in_index = true;
      out.push(line);
    } else {
      in_index = false;
    }
  } else if (line.startsWith('ALTER TABLE')) {
    const tableName = line.split('"')[1];
    if (tables.includes(tableName)) {
      in_fk = true;
      out.push(line);
    } else {
      in_fk = false;
    }
  } else {
    if (in_table || in_index || in_fk) {
      out.push(line);
      const trimmed = line.trim();
      if (trimmed === ');' || trimmed === ';') {
        in_table = false;
        in_index = false;
        in_fk = false;
      }
    }
  }
}

const dir = path.join('prisma', 'migrations', '20260804180000_init_marketing');
if (!fs.existsSync(dir)) {
  fs.mkdirSync(dir, { recursive: true });
}
fs.writeFileSync(path.join(dir, 'migration.sql'), out.join(''), 'utf8');

File: extract.py
tables = ['MarketingContact', 'MarketingList', 'MarketingContactList', 'MarketingTag', 'MarketingContactTag', 'MarketingSegment', 'EmailTemplate', 'CampaignMetricDaily', 'CampaignMetricHourly', 'DomainProviderMetric', 'AudienceMetricDaily', 'LinkMetric', 'DeliverabilityAlert', 'MtaNode', 'MtaProviderState']
with open('full_schema.sql', 'r', encoding='utf-8') as f:
    lines = f.readlines()

out = []
in_table = False
in_index = False
in_fk = False

for line in lines:
    if line.startswith('CREATE TABLE'):
        table_name = line.split('"')[1]
        if table_name in tables:
            in_table = True
            out.append(line)
        else:
            in_table = False
    elif line.startswith('CREATE UNIQUE INDEX') or line.startswith('CREATE INDEX'):
        table_name = line.split('"')[3]
        if table_name in tables:
            in_index = True
            out.append(line)
        else:
            in_index = False
    elif line.startswith('ALTER TABLE'):
        table_name = line.split('"')[1]
        if table_name in tables:
            in_fk = True
            out.append(line)
        else:
            in_fk = False
    else:
        if in_table or in_index or in_fk:
            out.append(line)
            if line.strip() == ');' or line.strip() == ';':
                if in_table:
                    in_table = False
                elif in_index:
                    in_index = False
                elif in_fk:
                    in_fk = False

import os
os.makedirs('prisma/migrations/20260804180000_init_marketing', exist_ok=True)
with open('prisma/migrations/20260804180000_init_marketing/migration.sql', 'w', encoding='utf-8') as f:
    f.writelines(out)
